apps/server/src/controllers/category.controller.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, Response, NextFunction } from 'express';
+import {
+  CreateCategoryInput,
+  UpdateCategoryInput,
+} from '../schemas/category.schema';
+import { CategoryService } from '../services/category.service';
+
+export class CategoryController {
+  private service: CategoryService;
+
+  constructor(_service: CategoryService) {
+    this.service = _service;
+  }
+
+  async getCategories(req: Request, res: Response, next: NextFunction) {
+    try {
+      const limit = Number(req.query['limit']);
```

Comment:
give it an initial values if user did not provide limit or offset:
const limit = Number(req.query['limit']) | 10;
const offset = Number(req.query['offset']) | 0;